universal-application-tool-0.0.1/app/repository/ApplicantRepository.java

```diff
@@ -32,6 +40,26 @@ public ApplicantRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exe
         () -> ebeanServer.find(Applicant.class).setId(id).findOneOrEmpty(), executionContext);
   }
 
+  public CompletionStage<Optional<Applicant>> lookupApplicant(String emailAddress) {
+    return supplyAsync(
+        () -> {
+          Optional<Account> account =
+              ebeanServer
+                  .find(Account.class)
+                  .where()
+                  .eq("email_address", emailAddress)
+                  .findOneOrEmpty();
+          if (account.isEmpty()) {
+            return Optional.empty();
+          }
+          // Return the applicant which was most recently created.
+          return account.get().getApplicants().stream()
```

Comment:
(blocking): Instead of checking `isEmpty` and then doing `.get().....` you could use

`return accountMaybe.flatMap(account -> account.getApplicants().stream().max(...));`